File: yunju/2025_12_Wk3/BAEK_9205_맥주마시면서걸어가기/solution.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <cmath>

using namespace std;

int T;
int n;

int cal_dist(pair<int, int> a, pair<int, int> b) {
    return abs(a.first-b.first) + abs(a.second-b.second);
}

void solve() {
    cin >> n;
    pair<int, int> home;
    cin >> home.first >> home.second;

    // 사이즈 n 짜리
    // 편의점들의 좌표가 담긴 vector
    vector<pair<int, int>> conv(n);

    for (int i=0; i<n; i++) {
        cin >> conv[i].first >> conv[i].second;
    }

    pair<int, int> festival;
    cin >> festival.first >> festival.second;

    vector<bool> visited(n, false);

    queue<pair<int, int>> q;
    q.push(home);

    while (!q.empty()) {
        pair<int, int> curr = q.front();
        q.pop();

        if (cal_dist(curr, festival) <= 1000) {
            cout << "happy" << endl;
            return;
            
        } 
        for (int i=0; i<n; i++) {
            if (!visited[i] && cal_dist(curr, conv[i]) <= 1000) {
                visited[i] = true;
                q.push(conv[i]);
            }
        }
    }
    cout << "sad" << endl;
    
}
int main() {
    cin >> T;
    while (T--) {
        solve();
    }
    return 0;
}

File: yunju/2025_12_Wk3/BAEK_9205_맥주마시면서걸어가기/Solution.java
import java.util.*;

class Point {
    int x, y;
    Point(int x, int y) {
        this.x = x;
        this.y = y;
    }
}
class Solution {
    public static int calDist(Point p1, Point p2) {
        return Math.abs(p1.x-p2.x) + Math.abs(p1.y-p2.y);
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            int n = sc.nextInt();
            Point home = new Point(sc.nextInt(), sc.nextInt());

            Point[] conv = new Point[n];

            for (int i=0; i<n; i++) {
                conv[i] = new Point(sc.nextInt(), sc.nextInt());
            }
            Point festival = new Point(sc.nextInt(), sc.nextInt());

            boolean[] visited = new boolean[n];
            Queue<Point> q = new LinkedList<>();
            q.add(home);

            boolean success = false;

            while (!q.isEmpty()) {
                Point curr = q.poll();

                if (calDist(curr, festival) <= 1000) {
                    success = true;
                    break;
                }

                for (int i=0; i<n; i++) {
                    if (!visited[i] && calDist(curr, conv[i]) <= 1000) {
                        visited[i] = true;
                        q.add(conv[i]);
                    }
                }
            }
            System.out.println(success ? "happy" : "sad");
            
        }
    }
}